### ISSUE TITLE
 AssertionError when running `meta-aggregate`

### ISSUE THREAD

**Issue by jsheunis** [datalad/datalad-metalad#143]
I have run `meta-extract | meta-add` on a number of subdatasets of a superdataset. The next step was to aggregate these metadata to the superdataset level. For this I ran, e.g.:

```
datalad meta-aggregate -d /Users/jsheunis/Documents/psyinf/Data/super /Users/jsheunis/Documents/psyinf/Data/super/bneuro/Mueller14_EEGSZ
```
which yields an error:
```
[ERROR  ] AssertionError(write_out: no destination available for <dataladmetadatamodel.metadata.Metadata object at 0x7fb92d492370>) (AssertionError)
```

If I do `meta-dump` on this subdataset, the metadata outputs as expected.

**Comment by jsheunis** [comment]
Interestingly, if I run the `meta-aggregate` command as part of a loop in my shell, it doesn't print the error, but the meta-data is not aggregated to the superdataset level:

```
.
.
.
/Users/jsheunis/Documents/psyinf/Data/super/bneuro/Mueller14_EEGSZ
meta_aggregate(ok): /Users/jsheunis/Documents/psyinf/Data/super [aggregated metadata into /Users/jsheunis/Documents/psyinf/Data/super]
.
.
.
```

**Comment by jsheunis** [comment]
Run with `--dbg` flag:

```
(catalog-v2) ➜  super git:(master) ✗ datalad --dbg meta-aggregate -d /Users/jsheunis/Documents/psyinf/Data/super /Users/jsheunis/Documents/psyinf/Data/super/bneuro/Mueller14_EEGSZ

Traceback (most recent call last):
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/bin/datalad", line 8, in <module>
    sys.exit(main())
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad/cmdline/main.py", line 211, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad/interface/base.py", line 783, in call_from_parser
    ret = list(ret)
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad/interface/utils.py", line 396, in generator_func
    for r in _process_results(
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad/interface/utils.py", line 574, in _process_results
    for res in results:
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad_metalad/aggregate.py", line 224, in __call__
    perform_aggregation(
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad_metalad/aggregate.py", line 268, in perform_aggregation
    copy_uuid_set(
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/datalad_metalad/aggregate.py", line 340, in copy_uuid_set
    element.deepcopy(new_realm=destination_metadata_store))
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/mappableobject.py", line 110, in deepcopy
    result = self.deepcopy_impl(new_mapper_family, new_destination, **kwargs)
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/metadatarootrecord.py", line 59, in deepcopy_impl
    self.dataset_level_metadata.deepcopy(new_mapper_family,new_destination),
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/mappableobject.py", line 110, in deepcopy
    result = self.deepcopy_impl(new_mapper_family, new_destination, **kwargs)
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/metadata.py", line 311, in deepcopy_impl
    copied_metadata.write_out(new_destination)
  File "/Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/mappableobject.py", line 72, in write_out
    assert destination is not None, f"write_out: no destination available for {self}"
AssertionError: write_out: no destination available for <dataladmetadatamodel.metadata.Metadata object at 0x7f913248e370>

> /Users/jsheunis/opt/miniconda3/envs/catalog-v2/lib/python3.9/site-packages/dataladmetadatamodel/mappableobject.py(72)write_out()
-> assert destination is not None, f"write_out: no destination available for {self}"
```

**Comment by jsheunis** [comment]
Fixed in a132864
